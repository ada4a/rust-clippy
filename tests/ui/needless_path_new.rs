#![warn(clippy::needless_path_new)]

use std::fs;
use std::path::Path;

fn takes_path(_: &Path) {}

fn takes_path_and_impl_path(_: &Path, _: impl AsRef<Path>) {}

fn main() {
    fs::write(Path::new("foo.txt"), "foo"); //~ needless_path_new

    fs::copy(
        Path::new("foo"), //~ needless_path_new
        Path::new("bar"), //~ needless_path_new
    );

    // no warning
    takes_path(Path::new("foo"));

    // the paramater that _could_ be passed directly, was
    // the parameter that could't, wasn't
    takes_path_and_impl_path(Path::new("foo"), "foo");
}
