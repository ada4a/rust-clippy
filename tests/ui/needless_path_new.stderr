error: the expression enclosed in `Path::new` implements `AsRef<Path>`
  --> tests/ui/needless_path_new.rs:22:15
   |
LL |     fs::write(Path::new("foo.txt"), "foo");
   |               ^^^^^^^^^^^^^^^^^^^^ help: remove the enclosing `Path::new`: `"foo.txt"`
   |
   = note: `-D clippy::needless-path-new` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::needless_path_new)]`

error: the expression enclosed in `Path::new` implements `AsRef<Path>`
  --> tests/ui/needless_path_new.rs:25:9
   |
LL |         Path::new("foo"),
   |         ^^^^^^^^^^^^^^^^ help: remove the enclosing `Path::new`: `"foo"`

error: the expression enclosed in `Path::new` implements `AsRef<Path>`
  --> tests/ui/needless_path_new.rs:26:9
   |
LL |         Path::new("bar"),
   |         ^^^^^^^^^^^^^^^^ help: remove the enclosing `Path::new`: `"bar"`

error: the expression enclosed in `Path::new` implements `AsRef<Path>`
  --> tests/ui/needless_path_new.rs:33:9
   |
LL |         Path::new("bar"),
   |         ^^^^^^^^^^^^^^^^ help: remove the enclosing `Path::new`: `"bar"`

error: aborting due to 4 previous errors

